nixos/modules/services/cluster/k3s/default.nix

```
@@ -33,15 +33,40 @@ in
 
     serverAddr = mkOption {
       type = types.str;
-      description = lib.mdDoc "The k3s server to connect to. This option only makes sense for an agent.";
+      description = lib.mdDoc ''
+        The k3s server to connect to.
+
+        Servers and agents need to communicate each other. Read
+        [the networking docs](https://rancher.com/docs/k3s/latest/en/installation/installation-requirements/#networking)
+        to know how to configure the firewall.
+      '';
       example = "https://10.0.0.10:6443";
       default = "";
     };
 
+    clusterInit = mkOption {
+      type = types.bool;
+      default = cfg.role == "server";
+      defaultText = literalExpression ''cfg.role == "server"'';
```

```suggestion
      defaultText = ''false";
```
Why is it necessary otherwise?